cogs/ctf/ctf.py

```diff
@@ -1,281 +1,76 @@
-            await category_folder.set_permissions(
-                self.bot.user, view_channel=True, connect=True)
-            await category_folder.set_permissions(
-                ctx.guild.default_role, view_channel=False, connect=False)
-            await category_folder.set_permissions(
-                ctf_role, view_channel=True, connect=True)
-
-        for val in AUTOGENERATED_CHANNELS.values():
-            channel = await ctx.guild.create_text_channel(val, category=category_folder)
-            if val == AUTOGENERATED_CHANNELS['access']:
-                await channel.set_permissions(ctx.guild.default_role, send_messages=False, view_channel=False)
-                await channel.set_permissions(self.bot.user, send_messages=True, view_channel=True)
-                await channel.set_permissions(ctf_role, send_messages=False, view_channel=True)
-
-                embed = discord.Embed(
-                    title="🔔 Join this CTF.",
-                    description="Do this if you are joining as a guest or participate in CTFs on other teams frequently (and thus don't want the CTF role). If you are a UIUC student, please opt-in using /ctf optin.",
-                    colour=discord.Colour.dark_gold()
-                )
-
-                message = await channel.send(embed=embed)
-                await message.add_reaction('❌')
-                await message.add_reaction('✔️')
-
-            if val == AUTOGENERATED_CHANNELS['logs']:
-                await channel.set_permissions(ctx.guild.default_role, send_messages=False, view_channel=False)
-                await channel.set_permissions(self.bot.user, send_messages=True, view_channel=True)
-                await channel.set_permissions(ctf_role, send_messages=False, view_channel=True)
-        await ctx.guild.create_voice_channel(f"{name}-general", category=category_folder)
-        await ctx.guild.create_voice_channel(f"{name}-work-1", category=category_folder)
-        await ctx.guild.create_voice_channel(f"{name}-work-2", category=category_folder)
-        await ctx.send(f':white_check_mark: The CTF "{name}" has been created!')
-
-    @commands.Cog.listener()
-    async def on_reaction_add(self, reaction, user):
+    @subcommand(name={"description": "The name of the CTF"})
+    @interactions.slash_default_member_permission(interactions.Permissions.ADMINISTRATOR)
+    async def create(self, ctx: SlashContext, name: str):
+        '''Creates a forum for the CTF'''
+        if (ctx.guild == None):
+            await ctx.send(":x: Must be used inside a guild.")
+            return
 
-        access_chan = reaction.message.channel
-        logs_channel = discord.utils.get(
-            access_chan.category.text_channels, name=AUTOGENERATED_CHANNELS['logs'])
-        if not access_chan.category.name.startswith(CTF_PREFIX) or access_chan.name != AUTOGENERATED_CHANNELS['access'] or user.bot:
+        ctf_name = sanitize_name("ctf-"+name, 100)
```

Comment:
100 is a positional argument, please specify it `max_length=100`

Reply:
If you want `sanitize_name` to be called with keyword argument only, you can change its definition to `def sanitize_name(name: str, *, max_length=32)`